tests/Microsoft.Identity.Test.Integration/HeadlessTests/SilentAuthTests.cs

```diff
@@ -0,0 +1,94 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Test.Integration.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Security;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class SilentAuthTests
+    {
+        private static readonly string[] _scopes = { "User.Read" };
+
+
+        [TestMethod]
+        public async Task SilentAuth_ForceRefresh_Async()
+        {
+            var labResponse = LabUserHelper.GetDefaultUser();
+            var user = labResponse.User;
+
+            var pca = new PublicClientApplication(labResponse.AppId, "https://login.microsoftonline.com/organizations");
+
+            Trace.WriteLine("Part 1 - Acquire a token with U/P");
+            AuthenticationResult authResult = await pca
+                .AcquireTokenByUsernamePassword(_scopes, user.Upn, new NetworkCredential("", user.Password).SecurePassword)
+                .ExecuteAsync(new CancellationTokenSource().Token)
+                .ConfigureAwait(false);
+            MsalAssert.AssertAuthResult(authResult, user);
+            var expiration1 = authResult.ExpiresOn;
+            // If test fails with "user needs to consent to the application, do an interactive request" error - see UsernamePassword tests
+
+            Trace.WriteLine("Part 2 - Acquire a token silently, with forceRefresh = true");
+            IAccount account = await MsalAssert.AssertSingleAccountAsync(labResponse, pca, authResult).ConfigureAwait(false);
+
+            authResult = await pca.AcquireTokenSilent(_scopes, account)
+                .WithForceRefresh(true)
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+            MsalAssert.AssertAuthResult(authResult, user);
+            var expiration2 = authResult.ExpiresOn;
+
+            // MSAL computes expiration at second granularity, so pause for 1 sec to observe the difference
+            Thread.Sleep(1000); 
```

Comment:
>Thread.Sleep(1000);  [](start = 12, length = 20)

can't we find a trigger to hook into instead of the sleep?

Reply:
if you _do_ need a sleep, please use await Task.Delay(1000) to keep the threading models consistent.

---
In reply to: [261296467](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/913#discussion_r261296467) [](ancestors = 261296467)